src/blockdata/transaction.rs

```diff
@@ -362,6 +383,42 @@ impl Transaction {
         sha256d::Hash::hash(&raw_vec)
     }
 
+    /// Returns the fee for the transaction by substracting the sum of the output amounts from the
+    /// provided set of transactions `spent_txs` outputs matching the transaction inputs. If some of
+    /// the provided transaction outputs inputs is not matching any of the transaction inputs than
+    /// the `UnmatchedTxForInput` error is returned. If a transaction matching current transaction
+    /// input does not have a corresponding output number, then a `UnmatchedVoutForInput` error is
+    /// returned. If some of the transaction outputs are not covered by any outputs within the
+    /// provided `spent_txs`, then `UnmatchedInput` error is returned. Otherwise, the returned value
+    /// is the amount of the transaction fee in satoshis.
+    pub fn get_fee(&self, spent_txs: Vec<Transaction>) -> Result<u64, Error> {
```

Comment:
Since only the outputs are being used, would it be better to takes only outputs as arguments? In some context, someone might not store  whole `Transaction`s. Maybe this should be `spent_txos: HashMap<(TxHas, u16), TxOutput>` instead?

Reply:
Certainly this should not be taking a vector by value like this. It could take an `IntoIter<&Transaction>`

Reply:
The function may return `Amount` possibly instead of `u64` 

(https://github.com/rust-bitcoin/rust-bitcoin/blob/f6f4ea79d6d11c2b4b91906def0884c1e760a6ad/src/util/amount.rs#L275)

Reply:
+ for Amount, + for pattern